sympy/utilities/lambdify.py

```diff
@@ -450,6 +455,33 @@ def wrapper(*argsx, **kwargsx):
                     "Expression:\n\n{expr}").format(sig=sig, expr=expr_str)
     return func
 
+def should_dummify(expr):
+    """
+    Returns the value of dummify to be used. This function either returns True or False.
+
+    Parameter:
+
+    expr : Represents the expression for which dummify should be checked
+
+    >>> from sympy.utilities.lambdify import should_dummify
+    >>> from sympy.abc import x, y
+    >>> should_dummify(r'32v2?g#Gmw845h$Wb53wi\phi')
+    True
+    >>> should_dummify('elif')
+    True
+    >>> should_dummify(x)
+    False
+
+    """
+    from sympy.printing.str import sstr
+    if keyword.iskeyword(sstr(expr)):
```

Comment:
@asmeurer , is there any problem in using sstr ? Also I tried to pass the printer from lambdify to should_dummify but I'm getting errors. One more thing to consider is that this function is public so is it okay to pass a printer as an argument of a public function ? Please clarify